partiql-tests-data/eval/rfc/0007.ion

```diff
@@ -0,0 +1,224 @@
+                      UNION ALL
+                      SELECT * FROM engineering.employees''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { name: "Bob Smith", title: "Consultant" },
+                { name: "Susan Smith", title: "Manager" },
+                { name: "Jane Smith",  title: "DEI" },
+                { name: "Manager", title: "Paul McCharmly" },
+                { name: "SDE", title: "George Parasol" },
+                { name: "Principal SDE", title: "Ringo Stone" },
+                { name: "Intern", title: "Eric Lennon" }
+            ]
+        }
+    },
+    {
+        name: "Example 2.2 — Union of Compatible Relations; Mismatch Column Names; With CORRESPONDING Clause",
+        statement: '''SELECT * FROM hr.employees
+                      UNION ALL CORRESPONDING BY ( name, title )
+                      SELECT * FROM engineering.employees''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { name: "Bob Smith", title: "Consultant" },
+                { name: "Susan Smith", title: "Manager" },
+                { name: "Jane Smith",  title: "DEI" },
+                { name: "Paul McCharmly", title: "Manager" },
+                { name: "George Parasol", title: "SDE" },
+                { name: "Ringo Stone", title: "Principal SDE" },
+                { name: "Eric Lennon", title: "Intern" }
+            ]
+        }
+    },
+    {
+        name: "Example 3 — Union of Heterogenous Relations",
+        statement: '''SELECT * FROM hr.employees               -- T1
+                      UNION
+                      SELECT title FROM engineering.employees  -- T2''',
+        assert: {
+            evalMode:[EvalModeCoerce, EvalModeError],
+            result: EvaluationFail,
+        }
+    },
+    {
+        name: "Example 4 — Intersection of Compatible Relations",
+        statement: '''SELECT title FROM hr.employees INTERSECT SELECT title FROM engineering.employees''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { title: "Manager" },
+            ]
+        }
+    },
+    {
+        name: "Example 5 — Difference of Compatible Relations",
+        statement: '''SELECT title FROM engineering.employees EXCEPT (
+                        SELECT title FROM hr.employees
+                        UNION
+                        SELECT title FROM accounting.employees
+                      )''',
+        assert: {
+            evalMode: [EvalModeCoerce, EvalModeError],
+            result: EvaluationSuccess,
+            output: $bag::[
+                { title: "SDE" },
+                { title: "Principal SDE" },
+            ]
+        }
+    },
+    {
+        name: "Example 6 — Value Coercion",
```

Comment:
nit: could make more clear from the test name that there's an error. The RFC says "-- ERROR! Not union-compatible" so perhaps "Example 6 - Value Coercion not union-compatible"?